scripts/build_coverage.sh

```diff
@@ -88,10 +93,27 @@ source "$CHIP_ROOT/scripts/activate.sh"
 
 # Generates ninja files
 if [ "$skip_gn" == false ]; then
-    gn --root="$CHIP_ROOT" gen "$OUTPUT_ROOT" --args='use_coverage=true'
+    gn --root="$CHIP_ROOT" gen "$OUTPUT_ROOT" --args='use_coverage=true chip_build_all_clusters_app=true'
+    ninja -C "$OUTPUT_ROOT"
+fi
+
+# Run unit tests
+if [ "$skip_gn" == false ]; then
     ninja -C "$OUTPUT_ROOT" check
 fi
 
+# Run yaml tests
+if [ "$run_yaml" == true ]; then
+    scripts/run_in_build_env.sh \
+        "./scripts/tests/run_test_suite.py \
+         --chip-tool ""$OUTPUT_ROOT/chip-tool \
+         run \
+         --iterations 1 \
+         --test-timeout-seconds 120 \
+         --all-clusters-app ""$OUTPUT_ROOT/chip-all-clusters-app
```

Comment:
Add a TODO to build other apps and tests?
I believe we have several tests that are supposed to run with other tools (like lock tests only pass with lock app)

Reply:
That is my plan, currently, only gcno files get generated by running the example apps, no gcda files generated, but gcda files are generated properly by running chip-tool, I guess there are some compiler difference between chip-tool and example apps. 
https://github.com/project-chip/connectedhomeip/issues/22931